src/entity.ts

```diff
@@ -898,22 +904,26 @@ export namespace entity {
             excludePathFromEntity(entity, newPath);
           });
         } else {
-          if (hasWildCard && remainderPath === '*') {
-            const parentEntity = entity.properties![firstPathPart].entityValue;
-
-            if (parentEntity) {
-              Object.keys(parentEntity.properties).forEach(path => {
-                const newPath = parentEntity.properties[path].arrayValue
-                  ? path + '[].*'
-                  : path + '.*';
-                excludePathFromEntity(parentEntity, newPath);
-              });
+          if (typeof entity.properties![firstPathPart] !== 'undefined') {
+            if (hasWildCard && remainderPath === '*') {
```

Comment:
I believe this can be simplified to:

```suggestion
          if (typeof entity.properties![firstPathPart] !== 'undefined' && hasWildCard && remainderPath === '*') {
```

Reply:
The problem is that the latter if has an else statement.

Reply:
I added this change an changed the else to an if else so that the indenting doesn't make it so difficult to read.